### PR TITLE
feat: exporting Node Bootstrapping Contract for external use

### PR DESCRIPTION
**Pull request by Devinwong**
**What type of PR is this?**

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/nbcontracthelper/helper.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package nbcontracthelper
+
+const (
+	defaultLinuxUser = "azureuser"
+)
+
+func GetDefaultLinuxAdminUsername() string {
+	// var abc nbc.ApiServerConfig
```
Comment (anujmaheshwari1) [inline pkg/nbcontracthelper/helper.go:8]: what is var abc?

Comment (Devinwong) [reply pkg/nbcontracthelper/helper.go:8]: sorry this was generated by copilot. I should have removed it 🤣

**Review by anujmaheshwari1 — APPROVED** [review]

File: pkg/proto/readme.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Notes for protobuf
+
+1. Run this command to compile .proto to .pb.go file.
+```/AgentBaker/hack/tools/bin/buf generate -o .```
```
Comment (cameronmeissner) [inline pkg/proto/readme.md:4]: nit: add a target to root-level Makfile instead?

File: pkg/proto/readme.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+1. If you haven't installed buf yet, run the following command, assumed /usr/local/bin is where your Go binary locates.
+
+```GO111MODULE=on GOBIN=/usr/local/bin go install github.com/bufbuild/buf/cmd/buf@v1.27.2```
```
Comment (anujmaheshwari1) [inline pkg/proto/readme.md:8]: Devin, we already have a make target to create the buf binary, https://github.com/Azure/AgentBaker/blob/c6fb221ae595a481af6380d777969916e5367634/hack/tools/Makefile#L25

File: pkg/proto/readme.md (hunk 2 above)
Comment (Devinwong) [reply pkg/proto/readme.md]: Updated the makefile. Do you mean that?

File: pkg/proto/readme.md (hunk 3 above)
Comment (Devinwong) [reply pkg/proto/readme.md]: updated to use that binary.

File: Makefile [hunk 4]
```diff
@@ -96,6 +100,8 @@ generate: bootstrap
 	popd \
 	)
 	GENERATE_TEST_DATA="true" go test ./pkg/agent...
+	@echo "compiling proto files in ./pkg/proto/ to pb.go files"
```
Comment (cameronmeissner) [inline Makefile:103]: do we actually need to do this on every `make generate`? I might lean towards keeping it as a separate target that we only run when changing those protos

File: go.mod [hunk 5]
```diff
@@ -19,6 +19,7 @@ require (
 )
 
 require (
+	github.com/Azure/AgentBaker/pkg/proto/gen/nbcontractv1 v0.0.0-20231024012513-889e5d49d683 // indirect
```
Comment (cameronmeissner) [inline go.mod:22]: a bit confused why we're now requiring this

File: Makefile (hunk 4 above)
Comment (Devinwong) [reply Makefile:103]: Makes sense. So we will run "make compile-proto-files" manually when we think needed

File: go.mod (hunk 5 above)
Comment (Devinwong) [reply go.mod]: You are right. This is not needed. Ran "go mod tidy" to clean it

**Review by cameronmeissner — APPROVED** [review]

**Pull request merged**